### PR TITLE
[CDF-986] XSS vulnerability in Date Input Component

### PR DESCRIPTION
**Pull request by Advent51**
- input value sets through jquery

### CODE REVIEW

**Comment by Advent51** [comment]
@afrjorge @annapopova1 @mchen-len-son @pamval please review

File: core-js/src/main/javascript/cdf-legacy/components/input.js [hunk 1]
```diff
@@ -635,7 +635,9 @@ var DateInputComponent = BaseComponent.extend({
 
     myself.placeholder()
         .addClass('date-input-container')
-        .html('<input class="date-input" id="' + inputId + '" value="' + inputValue + '"/>');
+        .html('<input class="date-input" id="' + inputId + '"/>');
+
+      $('#' + inputId).val(inputValue);
```
Comment (annapopova1) [inline core-js/src/main/javascript/cdf-legacy/components/input.js]: looks like incorrect indent here, should be 4 spaces for this line.

Comment (Advent51) [reply core-js/src/main/javascript/cdf-legacy/components/input.js]: Thanks!
Now it should be fixed.

File: core-js/src/main/javascript/cdf/components/DateInputComponent.js [hunk 2]
```diff
@@ -47,7 +47,9 @@ define([
 
       myself.placeholder()
         .addClass('date-input-container')
-        .html('<input class="date-input" id="' + inputId + '" value="' + inputValue + '"/>');
+        .html('<input class="date-input" id="' + inputId + '"/>');
+
+      $('#' + inputId).val(inputValue);
```
Comment (afrjorge) [inline core-js/src/main/javascript/cdf/components/DateInputComponent.js]: Using a selector to search for an element that is being created here seems to be excessive and might affect performance.

Comment (annapopova1) [reply core-js/src/main/javascript/cdf/components/DateInputComponent.js]: @afrjorge we can't do this in another way. And we do absolutely the same for text input component

Comment (afrjorge) [reply core-js/src/main/javascript/cdf/components/DateInputComponent.js]: I don't understand. Is there any reason for the following not to work?
```javascript
var $el = $('<input class="date-input" id="' + inputId + '"/>');
$el.val(inputValue);

myself.placeholder()
  .addClass('date-input-container')
  .html($el);
```

**Comment by Advent51** [comment]
@mbatchelor @pamval please review

**Pull request merged**